### ISSUE TITLE
Earth does not exist?

### ISSUE THREAD

**Issue by mauromortier** [teslamate-org/teslamate#4290]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Trying to up the compose file in portainer (stack) seems to work fine for grafana etc etc but not for teslamate-teslamate-1 and shows me the following in logs:
2024-10-20 23:38:10.732 [info] create index geofences__earth_box_ll_to_earth_latitude__longitude___radius_index

** (Postgrex.Error) ERROR 42704 (undefined_object) type "earth" does not exist

### Relevant log output

```shell
2024-10-20 23:41:16.507 [info] == Running 20190925152807 TeslaMate.Repo.Migrations.CreateGeoExtensions.change/0 forward

2024-10-20 23:41:16.509 [info] execute "CREATE EXTENSION IF NOT EXISTS cube"

2024-10-20 23:41:16.514 [info] execute "CREATE EXTENSION IF NOT EXISTS earthdistance"

2024-10-20 23:41:16.516 [info] create index geofences__earth_box_ll_to_earth_latitude__longitude___radius_index

** (Postgrex.Error) ERROR 42704 (undefined_object) type "earth" does not exist

    (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1

    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2

    (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1154: Ecto.Adapters.SQL.execute_ddl/4

    (ecto_sql 3.10.2) lib/ecto/migration/runner.ex:327: Ecto.Migration.Runner.log_and_execute_ddl/3

    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2

    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2

    (ecto_sql 3.10.2) lib/ecto/migration/runner.ex:290: Ecto.Migration.Runner.perform_operation/3
```

### Type of installation

Docker

### Version

latest

**Comment by ab300819** [comment]
I have the same issue

```log
2024-10-21 06:25:14.544 [info] create index geofences__earth_box_ll_to_earth_latitude__longitude___radius_index
** (Postgrex.Error) ERROR 42704 (undefined_object) type "earth" does not exist
    (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1
    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2
    (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1154: Ecto.Adapters.SQL.execute_ddl/4
    (ecto_sql 3.10.2) lib/ecto/migration/runner.ex:327: Ecto.Migration.Runner.log_and_execute_ddl/3
    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2
    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2
    (ecto_sql 3.10.2) lib/ecto/migration/runner.ex:290: Ecto.Migration.Runner.perform_operation/3
```

**Comment by mikkelbh** [comment]
Same issue here

**Comment by mauromortier** [comment]
When changing the compose to use Postgres 16, i get the notification that library is set up for 17 or something.

**Comment by JakobLichterfeld** [comment]
On a new install, please do a `docker volume rm "$(basename "$PWD")_teslamate-db"`

Or post your complete output.

**Comment by mauromortier** [comment]
that fixed it, thanks

**Comment by JakobLichterfeld** [comment]
> that fixed it, thanks

Apologies for the shaky start. Normally it is best to have the latest docs already published even before a new version release. This time it was very unfortunate and we will improve on this end.

**Comment by ab300819** [comment]
when i use postgres:16, it worked, but some query error appeared as shown in the screenshot.

<img width="1638" alt="image" src="https://github.com/user-attachments/assets/c04e48ae-871c-4091-9026-68480bf58cfa">

**Comment by JakobLichterfeld** [comment]
> when i use postgres:16, it worked, 

Perfect.

> but some query error appeared as shown in the screenshot.

You do not have enough data collected for the Battery Health Dashboard. The error message in this case is not the best.

**Comment by ab300819** [comment]
Thanks, I'll keep an eye on it.

**Comment by JakobLichterfeld** [comment]
resolved with v1.31.0
